Document dependencies required for developing Rust and C++ apps with Slint on macOS, Windows, and Linux (X11/Wayland/KMS)

We should document the libraries / packages required to develop with Slint (macOS/Windows/Linux). This needs to be included in the C++ and Rust documentation. For Node.js this belongs into a section about building from source.

We should cover package names for the most popular Linux distributions.

Relevant in this context is also the existing documentation about supported platforms:

    https://slint.dev/releases/1.3.2/docs/slint/src/introduction/supported_platforms

Perhaps it would make sense to document the dependencies in scope of this ticket in the same place and link to it from Rust/C++/Node.js.

Original:
```
  = note: /usr/bin/ld: cannot find -lxcb-render: No such file or directory
          /usr/bin/ld: cannot find -lxcb-shape: No such file or directory
          /usr/bin/ld: cannot find -lxcb-xfixes: No such file or directory
          collect2: error: ld returned 1 exit status
```

--- a/docs/astro/src/content/docs/guide/backends-and-renderers/backend_linuxkms.md
+++ b/docs/astro/src/content/docs/guide/backends-and-renderers/backend_linuxkms.md
@@ -15,6 +15,8 @@ and keyboard input.
  - libinput/libudev for input event handling from mice, touch screens, or keyboards.
  - libseat for GPU and input device access without requiring root access. (optional)
 
+## Dependencies
+
 For compilation, pkg-config is used to determine the location of the following required system libraries:
 
 | pkg-config package name | Package name on Debian based distros |
@@ -30,6 +32,8 @@ If you don't have `libseat` available on your target system, then instead of sel
 in exchange requires running the application as a user that's privileged to access all input and DRM/KMS device
 files; typically that's the root user.
 :::
+
+## Renderers
 
 The LinuxKMS backend supports different renderers. They can be explicitly selected for use through the
 `SLINT_BACKEND` environment variable.

--- a/docs/astro/src/content/docs/guide/backends-and-renderers/backend_winit.md
+++ b/docs/astro/src/content/docs/guide/backends-and-renderers/backend_winit.md
@@ -1,5 +1,6 @@
 ---
 <!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
+// cSpell: ignore libx libxcursor libxkbcommon
 title: Winit Backend
 description: Winit Backend
 next: false
@@ -33,3 +34,16 @@ The Winit backend reads and interprets the following environment variables:
 | Name               | Accepted Values | Description                                                        |
 |--------------------|-----------------|--------------------------------------------------------------------|
 | `SLINT_FULLSCREEN` | any value       | If this variable is set, every window is shown in fullscreen mode. |
+
+## Linux Dependencies
+
+On Linux, the Winit backend requires either X11 or Wayland to be available.
+Support of either can be enabled or disabled at compile time by setting the
+`backend-winit-x11` or `backend-winit-wayland` features (instead of `backend-winit`).
+
+For X11 the following runtime dependencies are required: libx11-xcb, xinput, libxcursor, libxkbcommon-x11, libx11.
+On Debian-based systems, these can be installed with:
+
+```sh
+sudo apt install libx11-xcb-dev xinput libxcursor-dev libxkbcommon-x11-dev libx11-dev
+```

